qa/tasks/lua_tests.py

```diff
@@ -0,0 +1,254 @@
+                        ],
+                    )
+
+@contextlib.contextmanager
+def configure(ctx, config):
+    assert isinstance(config, dict)
+    log.info('Configuring lua-tests...')
+    testdir = teuthology.get_testdir(ctx)
+    for client, properties in config['clients'].items():
+        (remote,) = ctx.cluster.only(client).remotes.keys()
+        luatests_conf = config['luatests_conf'][client]
+
+        conf_fp = BytesIO()
+        luatests_conf.write(conf_fp)
+        remote.write_file(
+            path='{tdir}/ceph/src/test/rgw/lua/lua-tests.{client}.conf'.format(tdir=testdir, client=client),
+            data=conf_fp.getvalue(),
+            )
+
+        remote.run(
+            args=[
+                'cd',
+                '{tdir}/ceph/src/test/rgw/lua'.format(tdir=testdir),
+                run.Raw('&&'),
+                './bootstrap',
+                ],
+            )
+
+    try:
+        yield
+    finally:
+        log.info('Removing lua-tests.conf file...')
+        testdir = teuthology.get_testdir(ctx)
+        for client, properties in config['clients'].items():
+            (remote,) = ctx.cluster.only(client).remotes.keys()
+            remote.run(
+                 args=['rm', '-f',
+                       '{tdir}/ceph/src/test/rgw/lua/lua-tests.{client}.conf'.format(tdir=testdir,client=client),
+                 ],
+                 )
+
+@contextlib.contextmanager
+def run_tests(ctx, config):
+    """
+    Run the lua tests after everything is set up.
+    :param ctx: Context passed to task
+    :param config: specific configuration information
+    """
+    assert isinstance(config, dict)
+    log.info('Running lua-tests...')
+    testdir = teuthology.get_testdir(ctx)
+    for client, client_config in config.items():
+        (remote,) = ctx.cluster.only(client).remotes.keys()
+
+        attr = ["!example_test"]
+
+        if 'extra_attr' in client_config:
+            attr = client_config.get('extra_attr')
+
+        args = [
+            'LUATESTS_CONF={tdir}/ceph/src/test/rgw/lua/lua-tests.{client}.conf'.format(tdir=testdir, client=client),
+            '{tdir}/ceph/src/test/rgw/lua/virtualenv/bin/python'.format(tdir=testdir),
+            '-m', 'nose',
+            '-s',
+            '{tdir}/ceph/src/test/rgw/lua/test_lua.py'.format(tdir=testdir),
+            '-v',
+            '-a', ','.join(attr),
+            ]
+
+        remote.run(
+            args=args,
+            label="lua tests"
+            )
+    yield
+
+@contextlib.contextmanager
+def task(ctx,config):
+    """
+
+    If you want to run the tests against your changes pushed to your remote repo you can provide 'suite_branch' and 'suite_repo'
+    parameters in your teuthology-suite command. Example command for this is as follows::
+
+    teuthology-suite --ceph-repo https://github.com/ceph/ceph-ci.git -s rgw:lua --ceph your_ceph_branch_name -m smithi --suite-repo https://github.com/your_name/ceph.git --suite-branch your_branch_name
+    
```

Comment:
It'd be great if you can add an example here of how we can use `extra_attr` in the yaml file.